contributors/devel/kubelet-cri-networking.md

```diff
@@ -0,0 +1,85 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Container Runtime Interface (CRI) Networking Specifications
+
+## Introduction
+[Container Runtime Interface (CRI)](container-runtime-interface.md) is 
+an ongoing project to allow container 
+runtimes to integrate with kubernetes via a newly-defined API. This document 
+specifies the network requirements for container runtime 
+interface (CRI). CRI networking requirements expand upon kubernetes pod 
+networking requirements. This document does not specify requirements 
+from upper layers of kubernetes network stack, such as `Service`. More 
+background on k8s networking could be found 
+[here](http://kubernetes.io/docs/admin/networking/)
+
+## Requirements
+1. Kubelet expects the runtime shim to manage pod’s network life cycle. Pod 
+networking should be handled accordingly along with pod sandbox operations. 
+   * `RunPodSandbox` must set up pod’s network. This includes, but is not limited 
+to allocating a pod IP, configuring the pod’s network interfaces and default 
+network route. Kubelet expects the pod sandbox to have an IP which is 
+routable within the k8s cluster, if `RunPodSandbox` returns successfully. 
+`RunPodSandbox` must return an error if it fails to set up the pod’s network. 
+If the pod’s network has already been set up, `RunPodSandbox` must skip 
+network setup and proceed. 
+   * `StopPodSandbox` must tear down the pod’s network. The runtime shim 
```

Comment:
StopPodSandbox gets called when kubelet decide the pod needs to be deleted. RemovePodSandbox gets called during kubelet GC. There is a lag in between. Kubelet expects the pod to stop running after StopPodSandbox. This means that all resources (cpu,memory,network) must be released after that. RemovePodSandbox just destroys the rest of the trace (container state, logs) of the pod